build.gradle

```diff
@@ -0,0 +1,143 @@
+plugins {
+    id 'org.ajoberstar.grgit' version '4.1.0'
+    id 'com.github.johnrengelman.shadow' version '6.1.0'
+    id 'java'
+}
+
+import org.apache.tools.ant.filters.ReplaceTokens
+import com.github.jengelman.gradle.plugins.shadow.tasks.ConfigureShadowRelocation
+
+allprojects {
+    group = 'net.ess3'
+    version = '2.18.1'
+}
+
+def commitsSinceLastTag() {
+    def tags = grgit.tag.list().stream().map({it.commit}).toList()
+    def commit = grgit.head()
+    def depth = 0
+    while (true) {
+        if (tags.contains(commit))
+            return depth
+        depth++
+        commit = grgit.resolve.toCommit(commit.parentIds.get(0))
+    }
+}
+
+ext {
+    GIT_COMMIT = grgit.head().abbreviatedId
+    GIT_DEPTH = commitsSinceLastTag()
+
+    checkstyleVersion = '8.36.2'
+    spigotVersion = '1.16.3-R0.1-SNAPSHOT'
+}
+
+artifacts {
+    tasks.shadowJar.outputs.files.each {
+        archives(it) {
+            builtBy shadowJar
+        }
+    }
+}
+
+task relocateShadowJar(type: ConfigureShadowRelocation) {
+    target = tasks.shadowJar
+}
+
+tasks.shadowJar.dependsOn tasks.relocateShadowJar
+
+subprojects {
+    apply plugin: 'java'
+    apply plugin: 'java-library'
+    apply plugin: 'checkstyle'
+
+    checkstyle {
```

Comment:
You should be able to just set the config directory here: https://github.com/KyoriPowered/indra/blob/master/indra-common/src/main/kotlin/net/kyori/indra/IndraCheckstylePlugin.kt#L42